aks-node-controller/parser/helper_test.go

```diff
@@ -886,6 +886,42 @@ func Test_getTargetEnvironment(t *testing.T) {
 			},
 			want: helpers.AksCustomCloudName,
 		},
+		{
+			name: "China location cluster config",
+			args: args{
+				v: &aksnodeconfigv1.Configuration{
+					CustomCloudConfig: &aksnodeconfigv1.CustomCloudConfig{},
+					ClusterConfig: &aksnodeconfigv1.ClusterConfig{
+						Location: "china",
+					},
+				},
+			},
+			want: "AzureChinaCloud",
+		},
+		{
+			name: "Germany location cluster config",
+			args: args{
+				v: &aksnodeconfigv1.Configuration{
+					CustomCloudConfig: &aksnodeconfigv1.CustomCloudConfig{},
+					ClusterConfig: &aksnodeconfigv1.ClusterConfig{
+						Location: "germanynortheast",
```

Comment:
im not sure german/dod clouds are actually something we'd ever see, though I'll take the added coverage regardless!